src/stores/ChannelStore.js

```diff
@@ -96,6 +96,21 @@ class ChannelStore extends DataStore {
    * @param {ChannelResolvable} channel The channel resolvable to resolve
    * @returns {?Snowflake}
    */
+
+  /**
+    * Obtains a channel from Discord, or the channel cache if it's already available.
+    * <warn>This is only available when using a bot account.</warn>
+    * @param {Snowflake} id ID of the channel
+    * @param {boolean} [cache=true] Whether to cache the new channel object if it isn't already
+    * @returns {Promise<GuildChannel>}
+    */
+  fetch(id, cache = true) {
+    const existing = this.get(id);
+    if (existing) return Promise.resolve(existing);
+
+    return this.client.api.channels(id).get().then(
+      data => this.client.guilds.fetch(data.guild_id).then(guild => this.add(data, guild, cache)));
```

Comment:
This will likely cause unexpected behaviour if you attempt to fetch a DM channel, since the guild ID will be undefined.  The API router would make a GET request to `/guilds`, which would give a useless and unexpected result.

Reply:
Yeah, I've also found that in the GuildStore if you use a Channel Snowflake it'll error with "Unauthorized Request". I have no idea how to fix this.